pkg/addon/config.go

```diff
@@ -0,0 +1,214 @@
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	elasticsearch "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const addonSecretName = "addon-cluster-logging-operator-parameters"
+
+var ErrNoAddonConfig = errors.New("No addon configiration found")
+
+type AddonConfiguration struct {
+	EsMaxRetention string `data-field:"es-max-retention" default:"7d"`
+	EsMemory       string `data-field:"es-memory" default:"2Gi"`
+	EsNodeCount    string `data-field:"es-node-count" default:"3"`
+	EsStorageClass string `data-field:"es-storage-class" default:"gp2"`
+	EsStorageSize  string `data-field:"es-storage-size" default:"200G"`
+}
+
+func ProcessAddonConfiguration(namespace string) {
+	// Verify if CLO's CR already present in the target namespace
+	cl, err := newClient()
+	if err != nil {
+		log.Error(err, "failed to create client")
+		return
+	}
+
+	present, err := verifyIfCrPresent(cl, namespace)
+	if err != nil {
+		log.Error(err, "unable to verify if clo's cr is present")
+		return
+	}
+
+	if present {
+		log.Info("clo's cr detected. Skipping addon configuration")
+		return
+	}
+
+	// Read addon secret from the target namespace
+	addonConfig, err := readAddonConfiguration(cl, namespace, addonSecretName)
+	if err != nil {
+		if err == ErrNoAddonConfig {
+			log.Info("no addon config found. skipping cr generation")
+			return
+		} else {
+			log.Error(err, "unable to read addon configuration")
+			return
+		}
+	}
+
+	// Generate new CLO's CR with parameters from the addon secret
+	cloCr, err := generateCrFromAddonConfig(addonConfig, namespace)
+	if err != nil {
+		log.Error(err, "unable to generate cr from addon config")
+		return
+	}
+
+	// Deploy the generated CR
+	err = cl.Create(context.Background(), cloCr)
+	if err != nil {
+		log.Error(err, "unable to create cr for addon config")
+	}
+	fmt.Println("addon configuration processed")
+}
+
+func verifyIfCrPresent(cl client.Client, namespace string) (bool, error) {
+	lg := &logging.ClusterLogging{}
+	err := cl.Get(context.Background(), client.ObjectKey{Name: "instance", Namespace: namespace}, lg)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func generateCrFromAddonConfig(addonConfig *AddonConfiguration, namespace string) (*logging.ClusterLogging, error) {
+	nodeCount, err := strconv.Atoi(addonConfig.EsNodeCount)
+	if err != nil {
+		return nil, fmt.Errorf("incorrect es-node-count parameter. %v", err)
+	}
+	mpEsStorage := resource.MustParse(addonConfig.EsStorageSize)
+
+	result := &logging.ClusterLogging{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "instance",
```

Comment:
can make the name `"instance"` a constant, it is used in `verifyIfCrPresent` also

Reply:
Fixed